diff -up Python-3.3.0b1/Lib/distutils/sysconfig.py.fix-CC-options.patch Python-3.3.0b1/Lib/distutils/sysconfig.py
--- Python-3.3.0b1/Lib/distutils/sysconfig.py.fix-CC-options.patch	2012-07-24 11:57:43.804353959 -0400
+++ Python-3.3.0b1/Lib/distutils/sysconfig.py	2012-07-24 11:59:38.341354843 -0400
@@ -580,7 +580,9 @@ def get_config_vars(*args):
             # skip checks if the compiler was overriden with a CC env variable
             if 'CC' not in os.environ:
                 cc = oldcc = _config_vars['CC']
-                if not find_executable(cc):
+                # CC might contain additional arguments e.g. "gcc -pthread", so
+                # look for the first word:
+                if not find_executable(cc.split()[0]):
                     # Compiler is not found on the shell search PATH.
                     # Now search for clang, first on PATH (if the Command LIne
                     # Tools have been installed in / or if the user has provided
