diff -up Python-3.3.1/Misc/NEWS.fix-test_gdb-test_threads Python-3.3.1/Misc/NEWS
diff -up Python-3.3.1/Tools/gdb/libpython.py.fix-test_gdb-test_threads Python-3.3.1/Tools/gdb/libpython.py
--- Python-3.3.1/Tools/gdb/libpython.py.fix-test_gdb-test_threads	2013-05-09 12:12:01.621592211 -0400
+++ Python-3.3.1/Tools/gdb/libpython.py	2013-05-09 12:12:01.632592209 -0400
@@ -1465,7 +1465,7 @@ class Frame(object):
         # This assumes the _POSIX_THREADS version of Python/ceval_gil.h:
         name = self._gdbframe.name()
         if name:
-            return name.startswith('pthread_cond_timedwait')
+            return 'pthread_cond_timedwait' in name
 
     def is_gc_collect(self):
         '''Is this frame "collect" within the garbage-collector?'''
