--- Python-3.1.1/Makefile.pre.in.install-tkinter-tests	2009-10-27 20:00:09.814539776 -0400
+++ Python-3.1.1/Makefile.pre.in	2009-10-27 20:01:43.460549947 -0400
@@ -835,6 +835,7 @@ EXTRAPLATDIR= @EXTRAPLATDIR@
 MACHDEPS=	$(PLATDIR) $(EXTRAPLATDIR)
 XMLLIBSUBDIRS=  xml xml/dom xml/etree xml/parsers xml/sax
 LIBSUBDIRS=	tkinter site-packages test test/output test/data \
+		tkinter/test tkinter/test/test_ttk tkinter/test/test_tkinter \
 		test/decimaltestdata \
 		encodings \
 		email email/mime email/test email/test/data \
