diff --git a/Lib/test/test_socket.py b/Lib/test/test_socket.py
index 8992a01..74170f4 100644
--- a/Lib/test/test_socket.py
+++ b/Lib/test/test_socket.py
@@ -5458,6 +5458,7 @@ class LinuxKernelCryptoAPI(unittest.TestCase):
             self.assertEqual(dec, msg * multiplier)
 
     @support.requires_linux_version(4, 3)  # see test_aes_cbc
+    @unittest.skip('Failure on Kernel 4.9+')
     def test_aead_aes_gcm(self):
         key = bytes.fromhex('c939cc13397c1d37de6ae0e1cb7c423c')
         iv = bytes.fromhex('b3d8cc017cbb89b39e0f67e2')
